core/controllers/ingest.py

```diff
@@ -107,15 +106,17 @@ def ingest(body: dict, excel_file: FileStorage) -> tuple[dict, int]:
                 ingest_dependencies.fund_specific_validation,
             )
         else:
-            # TODO https://dluhcdigital.atlassian.net/browse/SMD-653: replace hardcoded dependencies with dependency
-            #   injection
-            tables, p_error_messages = extract_process_validate_tables(workbook_data, PF_TABLE_CONFIG)
-            ct_error_messages = cross_table_validation(tables)
+            if not isinstance(ingest_dependencies, PFIngestDependencies):
+                raise ValueError("Ingest dependencies should be of type PFIngestDependencies")
+            tables, p_error_messages = extract_process_validate_tables(
+                workbook_data, ingest_dependencies.extract_process_validate_schema
+            )
+            ct_error_messages = ingest_dependencies.cross_table_validate(tables)
             error_messages = p_error_messages + ct_error_messages
             if error_messages:
                 raise ValidationError(error_messages)
             coerce_data(tables, PF_TABLE_CONFIG)
-            transformed_data = pf_r1_transform(tables, reporting_round)
+            transformed_data = ingest_dependencies.transform(tables, reporting_round)
```

Comment:
On the TF side this is called `transform_data`. Can we rationalise the interface - probably in the abstract base class?

Reply:
Good shout